crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -35,16 +35,25 @@ impl<'db> Unpacker<'db> {
         self.context.db()
     }
 
-    /// Unpack the value to the target expression.
     pub(crate) fn unpack(&mut self, target: &ast::Expr, value: UnpackValue<'db>) {
         debug_assert!(
             matches!(target, ast::Expr::List(_) | ast::Expr::Tuple(_)),
-            "Unpacking target must be a list or tuple expression"
+            "Unpacking target must be a list or tuple"
         );
-
         let mut value_ty = infer_expression_types(self.db(), value.expression())
```

Comment:
I don't think any of the changes in the first few lines here (or to the doc-comment) are necessary; let's revert them to keep the diff clean :-)